core/kotlinx-coroutines-core/test/CoroutineContextThreadLocalTest.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.experimental
+
+import org.junit.Test
+import kotlin.coroutines.experimental.*
+import kotlin.test.*
+
+class CoroutineContextThreadLocalTest : TestBase() {
+    @Test
+    fun testExample() = runTest {
+        val mainDispatcher = coroutineContext[ContinuationInterceptor]!!
+        val mainThread = Thread.currentThread()
+        val element = MyElement()
+        assertEquals(null, myThreadLocal.get())
+        val job = launch(element) {
+            assertTrue(mainThread != Thread.currentThread())
+            assertSame(element, coroutineContext[MyElement])
+            assertSame(element, myThreadLocal.get())
+            withContext(mainDispatcher) {
+                assertSame(mainThread, Thread.currentThread())
+                assertSame(element, coroutineContext[MyElement])
+                assertSame(element, myThreadLocal.get())
+            }
+            assertTrue(mainThread != Thread.currentThread())
+            assertSame(element, coroutineContext[MyElement])
+            assertSame(element, myThreadLocal.get())
+        }
+        assertEquals(null, myThreadLocal.get())
+        job.join()
+        assertEquals(null, myThreadLocal.get())
+    }
```

Comment:
```
 @Test
    fun testUndispatched()= runTest {
        val element = MyElement()
        val job = launch(context = CommonPool + element, start = CoroutineStart.UNDISPATCHED) {
            assertNotNull(myThreadLocal.get())
            yield()
            assertNotNull(myThreadLocal.get())
        }

        assertNull(myThreadLocal.get())
        job.join()
        assertNull(myThreadLocal.get())
    }
```

This test fails

Reply:
Fixed it. Added test. Turned out there was a combination of bugs. Not only the test did not work, but actually none of the tests worked because test resources were not included into test builds, but those failures were silently ignored.